keps/sig-windows/2802-identify-windows-pods-apiserver-admission/README.md

```diff
@@ -0,0 +1,388 @@
+  - [ ] (R) Ensure GA e2e tests for meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+The main goal of this enhancement is to authoritatively identify the Operating System
+on which a pod would like to run.
+
+
+## Motivation
+
+Identifying the OS of the pods during the API Server admission time is crucial to apply
+appropriate security constraints to the pod. Without proper identification, some admission
+plugins may apply unnecessary security constraints to the pods or in the
+worst case, don't apply security constraints at all.
+
+
+### Goals
+
+Admission plugins and/or Admission Webhooks identifying the Operating System on which
+pod intends to run.
+
+### Non-Goals
+
+
+- Validating if the pods have appropriate security constraints.
+- Interaction with container runtimes.
+
+## Proposal
+
+
+Identifying the Windows pods during the scheduling phase can be done
+in the following ways:
+- Based on [nodeSelector and tolerations](https://kubernetes.io/docs/setup/production-environment/windows/user-guide-windows-containers/#ensuring-os-specific-workloads-land-on-the-appropriate-container-host) in the pod spec with 
+  Windows node specific labels.
+- Based on [runtimeclasses](https://kubernetes.io/docs/setup/production-environment/windows/user-guide-windows-containers/#ensuring-os-specific-workloads-land-on-the-appropriate-container-host) in the pod spec
+
+The runtimeclass is a higher level abstraction which gets translated again to nodeSelectors+tolerations. While the 
+nodeSelector with reserved OS label is good enough, it can be viewed as an side-effect of scheduling constraints 
+and need not necessarily express the user intention fully. So, we propose to add a new field to the pod spec called 
```

Comment:
I think this field was primarily the result of the distinction between
1. pod targets a windows node
2. pod OS is [linux,windows]

Since a pod targeting a windows node could still be a linux pod, we needed a way to express scheduling distinct from the pod OS.  The pod OS informs what combination of fields can be specified and what values need to be enforced during pod security admission.